lib/xnvme_be_macos_driverkit_dev.c

```diff
@@ -0,0 +1,218 @@
+// SPDX-FileCopyrightText: Samsung Electronics Co., Ltd
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifdef XNVME_BE_MACOS_ENABLED
+#include <xnvme_be_macos_driverkit.h>
+#endif
+
+#include <libxnvme.h>
+#include <xnvme_be.h>
+#include <xnvme_be_nosys.h>
+#ifdef XNVME_BE_MACOS_ENABLED
+#include <mach/mach_error.h>
+#include <CoreFoundation/CoreFoundation.h>
+
+#include <xnvme_dev.h>
+
+int
+xnvme_be_macos_driverkit_enumerate(const char *sys_uri, struct xnvme_opts *opts,
+				   xnvme_enumerate_cb cb_func, void *cb_args)
+{
+	XNVME_DEBUG("xnvme_be_macos_driverkit_enumerate()");
+
+	if (sys_uri) {
```

Comment:
See notes on declaration / initialization order and error-handling.

Reply:
Done.